internal/content_coding.go

```diff
@@ -255,10 +255,17 @@ func (d *ZlibDecoder) Decode(data []byte) ([]byte, error) {
 	if err != nil {
 		return nil, err
 	}
-	_, err = io.Copy(d.buf, r)
-	if err != nil && !errors.Is(err, io.EOF) {
-		return nil, err
+
+	for {
+		_, err = io.CopyN(d.buf, r, 1024*1024)
+		if err != nil {
+			if errors.Is(err, io.EOF) {
+				break
+			}
+			return nil, err
+		}
 	}
+
```

Comment:
How is this different to the code before? You can still exhaust memory by a [so called ZIP bomb](https://en.wikipedia.org/wiki/Zip_bomb). The only valid defense is to provide a upper decompression limit, i.e. specify `n` value passed to `io.CopyN`.

Reply:
For tests it is easy, we can hardcode `n` value.

But `ContentDecoder`s from `internal/content_coding.go` are used in:
- AMQP Consumer Input Plugin
- Socket Listener Input Plugin

For these two new configuration field should be added (with reasonable default value) to hold upper size limit.

@srebhan Will that be ok?

Reply:
Yes I think that's OK. We should set this to unlimited somehow... @powersj any opinion?

Reply:
Sounds go to add, but how do we keep the unlimited? Maybe a setting of 0 == unlimited and if so we use `io.Copy`?

Reply:
Setting it to unlimited by default won't really fix the issue ;)

@srebhan @powersj Please, review current implementation.

Reply:
> Please, review current implementation.

Understood, but what risk do we run by putting in a new default and breaking users?

Reply:
@powersj For both plugins default was set to `500MB`. 
It will not change anything for users operate on payload of smaller size. 

Users with bigger payload will need to set larger value in config. We may call it "breaking users" or "fixing security gaps" ;)

Reply:
ok we chatted and agree with the new default, we will want to call this out in change log for the next release